src/usr/local/www/status_logs_filter.php

```diff
@@ -134,9 +162,156 @@ function getGETPOSTsettingvalue($settingname, $default) {
+$pconfig['logprivatenets'] = !isset($config['syslog']['nologprivatenets']);
+$pconfig['filterdescriptions'] = $config['syslog']['filterdescriptions'];
+
+$save_settings = getGETPOSTsettingvalue('save_settings', null);
+
+if ($save_settings) {
+
+	# All
+	$cronorder = getGETPOSTsettingvalue('cronorder',  null);
+	$nentries = getGETPOSTsettingvalue('nentries', null);
+	$logfilesize = getGETPOSTsettingvalue('logfilesize', null);
+	$format  = getGETPOSTsettingvalue('format',  null);
+
+	# System General (main) Specific
+	$loglighttpd  = getGETPOSTsettingvalue('loglighttpd',  null);
+
+	# Firewall Specific
+	$logdefaultblock = getGETPOSTsettingvalue('logdefaultblock',  null);
+	$logdefaultpass = getGETPOSTsettingvalue('logdefaultpass', null);
+	$logbogons = getGETPOSTsettingvalue('logbogons', null);
+	$logprivatenets  = getGETPOSTsettingvalue('logprivatenets',  null);
+	$filterdescriptions  = getGETPOSTsettingvalue('filterdescriptions',  null);
+
+	unset($input_errors);
+	$pconfig = $_POST;
+
+	/* input validation */
+	if (isset($nentries) && (strlen($nentries) > 0)) {
+		if (!is_numeric($nentries) || ($nentries < 5) || ($nentries > 2000)) {
+			$input_errors[] = gettext("Number of log entries to show must be between 5 and 2000.");
+		}
+	}
+
+	if (isset($logfilesize) && (strlen($logfilesize) > 0)) {
+		if (!is_numeric($logfilesize) || ($logfilesize < 100000)) {
+			$input_errors[] = gettext("Log file size must be numeric and greater than or equal to 100000.");
+		}
+	}
+
+	if (!$input_errors) {
+
+		# Clear out the specific log settings and leave only the applied settings to override the general logging options (global) settings.
+		unset($config['syslog'][$specific_log]);
+
+	# All
+		if ($cronorder != '') { # if not using the general logging options setting (global)
+			$config['syslog'][$specific_log]['cronorder'] = $cronorder;
+		}
+
+		if (isset($nentries) && (strlen($nentries) > 0)) {
+			$config['syslog'][$specific_log]['nentries'] = (int)$nentries;
+		}
+
+		if (isset($logfilesize) && (strlen($logfilesize) > 0)) {
+			$config['syslog'][$specific_log]['logfilesize'] = (int)$logfilesize;
+		}
+
+		if ($format != '') { # if not using the general logging options setting (global)
+			$config['syslog'][$specific_log]['format'] = $format;
+		}
+
+	# System General (main) Specific
+		if ($logfile == 'system') {
+			$oldnologlighttpd = isset($config['syslog']['nologlighttpd']);
+			$config['syslog']['nologlighttpd'] = $loglighttpd ? false : true;
+		}
+
+	# Firewall Specific
+		if ($logfile == 'filter') {
+			$oldnologdefaultblock = isset($config['syslog']['nologdefaultblock']);
+			$oldnologdefaultpass = isset($config['syslog']['nologdefaultpass']);
+			$oldnologbogons = isset($config['syslog']['nologbogons']);
+			$oldnologprivatenets = isset($config['syslog']['nologprivatenets']);
+
+			$config['syslog']['nologdefaultblock'] = $logdefaultblock ? false : true;
+			$config['syslog']['nologdefaultpass'] = $logdefaultpass ? true : false;
```

Comment:
Now I really do not understand - the logic here is also the reverse of what I expected.